File: LOGIC BUILDING/FACTORIAL.py
"""num = int(input("Enter number: "))
fact = 1
for i in range(1,num+1):
    fact *= i

print(fact)"""

def factorial(n):
    if n==0:
        return 1
    return n*factorial(n-1)

num = 5
print(f"The factorial of {num} is {factorial(num)}")

File: factorial.java
import java.util.Scanner;

public class factorial {
    public static int fact(int a){
        int product = 1;
        if (a>0){
            for(int i=1;i<=a;i++){
                product *= i;
            }}
        else if(a==0){
            return 1;
        }
        else{
            return -1;
        }
        return product;

    }
    public static void main(String args[])
    {
        Scanner sc = new Scanner(System.in);
        int a = sc.nextInt();
        System.out.println("The factorial of number: "+ fact(a));
    }
}
